handlers/update_user.go

```diff
@@ -0,0 +1,43 @@
+package handlers
+
+import (
+	"encoding/json"
+	"http-go/models"
+	"http-go/storage"
+	"io/ioutil"
+	"net/http"
+)
+
+func UpdateUser(src storage.Updater) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setupResponse(&w)
+
+		if r.Method != http.MethodPut {
+			http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+			return
+		}
+
+		user := models.User{}
```

Comment:
Same thing you can use `user := new(models.User)`. Look here for [https://github.com/speedandfunction-lavrik/go-http-training/pull/1/files#r565633116](https://github.com/speedandfunction-lavrik/go-http-training/pull/1/files#r565633116).